### ISSUE TITLE
Put prompt/Call to action text in another color.

### ISSUE THREAD

**Issue by IanMayo** [debrief/pepys-import#646]
## 🐞 Overview
We're currently formatting the prompt lines in **bold**.  They would be even easier to spot if they were in a different color.

Right now (*), the backdrop in MS-Windows & Jupyter is black.  So, for this issue the prompt should be light-blue.

(*) @robintw is experimenting with switching to a white background, which would encourage us to use a darker color for the prompt text.  But, any such change should only impact on the method that provides the formatted prompt text.

## 🔗 Feature
supports #595 

## 🔢 Acceptance criteria
* [ ] Prompts in `Pepys-import` shaded in light blue in local terminal
* [ ] Prompts in `Pepys-import` shaded in light blue in Jupyter Notebook

**Comment by BarisSari** [comment]
@IanMayo, should I change colour of answers, too? I changed colour of bold prompt lines and they look like:
<img width="712" alt="image" src="https://user-images.githubusercontent.com/41292272/100577020-5ab0ad00-32f0-11eb-8fc8-846a0f01d8c9.png">
<img width="712" alt="image" src="https://user-images.githubusercontent.com/41292272/100577079-7caa2f80-32f0-11eb-91fc-8f590370f786.png">
<img width="712" alt="image" src="https://user-images.githubusercontent.com/41292272/100577116-90559600-32f0-11eb-82b7-2776decf665e.png">

When I change also the answers, they look like:
<img width="712" alt="image" src="https://user-images.githubusercontent.com/41292272/100577306-0659fd00-32f1-11eb-8fef-7c99a01aebae.png">
<img width="712" alt="image" src="https://user-images.githubusercontent.com/41292272/100577268-e75b6b00-32f0-11eb-83dc-26c2af4027d5.png">
<img width="712" alt="image" src="https://user-images.githubusercontent.com/41292272/100577245-db6fa900-32f0-11eb-8a50-77ae6ab82094.png">

**Comment by IanMayo** [comment]
Oooh, this is exciting @BarisSari 

I welcome advice from @clenoble . When we were introducing bold text, I'm pretty sure she didn't want the multiple-choice answers in bold.  So, only the question is highlighted.  My guess is that Céline won't want the numbered answers in blue.

**Comment by clenoble** [comment]
@BarisSari @IanMayo I like it like that a lot. It makes more sense to make all Pepys instructions in blue and all user inputs + feedback in white. The main question is in bold so there is still an emphasis on it.

**Comment by IanMayo** [comment]
Ok @BarisSari - there we have it :-)

**Comment by robintw** [comment]
I think this won't work properly on Windows - from the investigations I've done into colours on the Windows Command Prompt, it seems that bold text isn't actually supported, and it just switches to a brighter white for the text colour to make it look more obvious (the default colours for the Command Prompt are actually light gray for text and black for the background). Therefore, if we switch to blue for everything we display and then add on bold on top for the questions, then on Windows I think it will all look the same.

I've finished work now for the day, but I will test this tomorrow on Windows and just check that I'm right. There are ways to do different 'stylesheets' for prompt_toolkit, so we could do different styles on Windows and Linux, but that will make it look different on the actual systems it's used on (Windows) and things like the Binder training notebooks (which are Linux).

**Comment by IanMayo** [comment]
Thanks for investigating that @robintw . I look forward to your further conclusions tomorrow.

But, @clenoble - I guess this will mean we're back to highlighting the question, but not the multiple choice questions.

**Comment by robintw** [comment]
This is how the blue text looks on Windows, with and without bold. You can see that the bold setting doesn't actually make it bold, it just brightens the colour:

<img width="474" alt="Screen Shot 2020-12-01 at 11 49 53" src="https://user-images.githubusercontent.com/296686/100737019-86549580-33cb-11eb-8dd9-068f29ebbf48.png">

If we look without any custom colour, then we find the same thing applies - the 'normal' text is in light gray, and using bold makes it brighter white:

<img width="482" alt="Screen Shot 2020-12-01 at 11 51 58" src="https://user-images.githubusercontent.com/296686/100737152-b13ee980-33cb-11eb-9930-55cc1385c7a2.png">

With a white background, they look like this:

<img width="364" alt="Screen Shot 2020-12-01 at 11 54 06" src="https://user-images.githubusercontent.com/296686/100737349-f4995800-33cb-11eb-9d1c-daf8fcf7349c.png">

and:

<img width="305" alt="Screen Shot 2020-12-01 at 11 53 42" src="https://user-images.githubusercontent.com/296686/100737359-fa8f3900-33cb-11eb-9ee2-6eed1733ff0a.png">

In the latter one, the bold text actually looks _less_ obvious because of the way the colours are working.

I think if we choose colours we need to decide at the same time whether we're going to go for a white background, as it will affect the colours we choose. I _think_ I've got a way of getting a white background working properly on Windows - I haven't yet experimented for the xterm in Jupyter on Binder. (The Windows way involves setting special properties on the shortcut, but there doesn't seem to be programmatic access to these - so we will have to provide a shortcut with the properties set, and then just copy it to the right place and change the place it points to - all do-able but a bit strange).

What do you think @clenoble and @IanMayo?

**Comment by robintw** [comment]
Oh, I forgot to say. We can use separate 'style-sheets' (like CSS) for different platforms if we want. To do that we define styles and use those in the FormattedText call, and then also pass a style argument - see https://python-prompt-toolkit.readthedocs.io/en/master/pages/advanced_topics/styling.html#class-names

**Comment by IanMayo** [comment]
Thanks @robintw .  I did think that we only need to consider MS-Windows styles.  But, I now remember how useful the `Binder` previews are.  
I think the perspective is that we're trying to reproduce the MS-Windows experience in Linux.  So, if we need style sheets to do that then we can.  My guess is that we probably don't - and our custom formatter is probably sufficient.

**Comment by IanMayo** [comment]
I think our strategy here is sensitive to if we can do white background in Binder/xterm.  I think we can use bold and color with black backdrop, only one of them with white backdrop.

Once we know that step, we know which color scheme we're using, and what formatting to support.

**Comment by clenoble** [comment]
The light "bold" blue on white background is hard to read. I would need to check all the other possible colors, but if they work in the same way and are also hard to read, a black background will be best

**Comment by robintw** [comment]
I've got a white background working in the Binder xterm terminal:

![image](https://user-images.githubusercontent.com/296686/100789384-495dc280-340e-11eb-8e6d-cb514e09f15b.png)

The downside of a white background there is that the terminal blends into the notebook, making it hard to see exactly what is terminal and what is notebook. A simple (but slightly messy) alternative is to leave the black surrounds:

![image](https://user-images.githubusercontent.com/296686/100789568-875ae680-340e-11eb-92fd-0637d149ce99.png)

Note that bold text works fine on a terminal with a white background on Linux, as bold is supported by the terminal and doesn't have to be 'emulated' by brightening the text.

So, in summary:
1. It is possible to have a terminal with a white background on both Windows and on Binder
2. If we use a white background on Windows then we can't use bold text
3. If we use a white background then we'll have to carefully choose colours to make sure they show up easily on a white background

I'll leave it up to you @IanMayo and @clenoble to decide what you want to do 😄

**Comment by IanMayo** [comment]
@robintw - would you mind pushing your branch to a draft PR, so we can make a playable demo in Binder please?

**Comment by clenoble** [comment]
@robintw @IanMayo Where can I find the available colors for white background in normal and so-called "bold" version? Will these colors be identical in both Windows and Binder?

**Comment by robintw** [comment]
@IanMayo I've pushed a PR with a Binder demo of a white background terminal. I've left the black border for the moment, but can remove it later if needed. Binder here: https://mybinder.org/v2/gh/debrief/pepys-import/binder-white-background?filepath=demos/, PR here: https://github.com/debrief/pepys-import/pull/656. To do this, I had to change notebook_xterm, so the only change in the Pepys PR is to install a different branch of our fork of notebook_xterm.

@clenoble: Unfortunately, it's complicated. Here's what I _think_ is correct:
 - Theoretically, any colour (specified as a HTML colour, like `#ffffff`) is available
 - However, Windows is notoriously picky about how it displays colours in the terminal, so we need to test on Windows to check they still look correct
 - On Linux, the bold version is exactly the same colour, but with actual bold text.
 - On Windows, prompt_toolkit does something itself to try and 'brighten' the colour and make it look more bold - I don't know exactly what logic it uses for that, and I wouldn't like to trust that it will always do the right thing, particularly when backgrounds are set differently. Therefore, I would suggest that we only use the 'bold' style if we're using a black background.
 - (Note from the above two points that anything with bold text will display differently on Windows and Linux)

The 'path of least complexity' would be to use a black background (which simplifies all the settings we'd need to configure to get a white background) and just use colours to denote different types of text (ie. not using bold at all). However, we could probably get something more complicated than that working - but it would look different on Linux and Windows.

**Comment by IanMayo** [comment]
Thanks for that @robintw 

I _guess_ we present these types of text:
1. Prompt
2. Options
3. Context-specific help guidance
4. Other general guidance (reminder about how to trigger help)
5. New section header text (`== Resolve Platform ==`)
6. User responses
7. Add new element summary (field-name)
8. Add new element summary (value)
9. Intro banner

Yes, using a unique color for all of the above would be a horror.  But, we may be able to categorise them, resulting in a smaller set of colors that still make sense logically.

Does that seem feasible @clenoble ?

**Comment by IanMayo** [comment]
@Baris, Celine expressed a preference for this styling of prompts & multiple-choice elements:
![image](https://user-images.githubusercontent.com/1108513/101172804-907bcb80-3639-11eb-92be-862413b46c28.png)

So, prompt in bold, choice elements in normal weight.

**Comment by BarisSari** [comment]
Okay, @IanMayo. It's how it looks like in the current version of the [PR](https://github.com/debrief/pepys-import/pull/648/files). I don't need to make an extra change. But as you know, Windows will show bold texts differently. I don't have a windows machine to test that. Maybe, Robin could help.

**Comment by robintw** [comment]
I'll test on Windows and post some screenshots here

**Comment by robintw** [comment]
This is how it looks on Windows @IanMayo @BarisSari @clenoble. Is it ok?

![image](https://user-images.githubusercontent.com/296686/101181120-6c71b780-3644-11eb-8039-40b75427b0cc.png)

**Comment by IanMayo** [comment]
Thanks @robintw - I'm happy to give it a go, and learn lessons from the working version.
